Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -170,26 +170,36 @@ private static IEnumerable<BusinessObjectValidationFailure> CreateBusinessObject
 
         if (markAsHandled)
         {
-          _unhandledValidationFailures.Remove((businessObjectValidationFailure, validationFailure));
-          _handledValidationFailures.Add(validationFailure);
+          MoveFailureToHandled(businessObjectValidationFailure, validationFailure);
         }
       }
 
       return result;
     }
 
-    public IReadOnlyCollection<UnhandledBusinessObjectValidationFailure> GetUnhandledValidationFailures (
-        IBusinessObject businessObject,
-        bool includePartiallyHandledFailures = false)
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailures (
+        [NotNull] IBusinessObject businessObject,
+        bool includePartiallyHandledFailures = false,
+        bool markAsHandled = false)
     {
-      ArgumentUtility.CheckNotNull("businessObject", businessObject);
+      ArgumentUtility.CheckNotNull(nameof(businessObject), businessObject);
 
-      return _unhandledValidationFailures
+      var validationFailures = _unhandledValidationFailures
           .Where(f => Equals(f.BusinessObjectValidationFailure.ValidatedObject, businessObject))
-          .Where(f => includePartiallyHandledFailures || !_handledValidationFailures.Contains(f.ValidationFailure))
-          .Select(f => f.BusinessObjectValidationFailure)
-          .Select(f => new UnhandledBusinessObjectValidationFailure(f.ErrorMessage, f.ValidatedProperty))
-          .ToArray();
+          .Where(f => includePartiallyHandledFailures || !_handledValidationFailures.Contains(f.ValidationFailure));
+
+      var result = new List<BusinessObjectValidationFailure>();
+      foreach (var (businessObjectValidationFailure, validationFailure) in validationFailures)
+      {
+        result.Add(businessObjectValidationFailure);
+
+        if (markAsHandled)
+        {
+          MoveFailureToHandled(businessObjectValidationFailure, validationFailure);
+        }
```

Comment:
```suggestion
        if (markAsHandled)
          MoveFailureToHandled(businessObjectValidationFailure, validationFailure);
```
Since we're touching it anyway... 
Suggestion, it just looks "cleaner" to me.